changelog/core-memory-__delete.dd

```diff
@@ -0,0 +1,25 @@
+`core.memory.__delete` has been added
+
+$(REF __delete, core, memory) allows easy migration from the deprecated `delete`.
+`__delete` behaves exactly like `delete` and will call the class dtor immediately
+and call the garbage collector to immediately release the memory:
+
+---
+class B
+{
+    string a;
+}
+B b = new B();
+b.a = "exists";
+auto s = b.a;
+
+__delete(b);
+assert(b is null);
+assert(s == "exists");
+---
+
+For example, one a Posix platform you can simply run:
+
+$(CONSOLE
+sed "s/delete \(.*\);/__delete(\1);/" -i **/*.d
+)
```

Comment:
I suggest adding something to the effect that this is intended to only be used as a last resort when `destroy` is not an option.  At least that's my understanding:  We want to migrate users to `destroy`.

Reply:
Potential verbage:  "Users should prefer `object.destroy` to explicitly finalize objects, and only resort to `__delete` when `object.destroy` would not be a feasible option."

Reply:
The sed command could be a bit refined: `s/\Wdelete[ \t]+\(.*\);/__delete(\1);/g`